homeassistant/helpers/template.py

```diff
@@ -864,10 +896,15 @@ def _collect_state(hass: HomeAssistant, entity_id: str) -> None:
         entity_collect.entities.add(entity_id)
 
 
+@lru_cache(maxsize=CACHED_TEMPLATE_STATES)
+def _template_state_no_collect(hass: HomeAssistant, state: State) -> TemplateState:
+    return TemplateState(hass, state, collect=False)
+
+
 def _state_generator(hass: HomeAssistant, domain: str | None) -> Generator:
```

Comment:
Not sure if it is worth the overhead here to keep this a generator: `async_all` is not a generator and `sorted` needs to check all entries so it can sort them. So it's all in memory.

Reply:
Oh right, I guess it won't instantiate `TemplateState` objects in that case 👍